gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala

```diff
@@ -54,6 +54,11 @@ trait BasicScanExecTransformer extends LeafTransformSupport with SupportFormat {
   // TODO: Remove this expensive call when CH support scan custom partition location.
   def getInputFilePaths: Seq[String]
 
+  def getLocalFilesNodes: Seq[(java.io.Serializable, Array[String])] =
```

Comment:
can we define a case class for `(java.io.Serializable, Array[String])`? To make `Serializable` as return type is so wired.

Reply:
This is because CK is using `ExtensionTable` instead of `LocalFilesNode`, and their common superclass is `Serializable`. Now, a new interface called `ReadSplit` is introduced to represent the abstraction of both classes. Additionally, `preferredLocations` have been included in the class to avoid using Tuple.